### PR TITLE
Make ignore_*::count() returns a size in # of elements

### PR DESCRIPTION
**Pull request by jfalcou**

### CODE REVIEW

File: include/eve/conditional.hpp [hunk 1]
```diff
@@ -130,12 +130,17 @@ namespace eve
       return mask;
     }
 
-    template<typename T> EVE_FORCEINLINE std::ptrdiff_t offset(eve::as_<T> const&) const
+    template<typename T> EVE_FORCEINLINE constexpr std::ptrdiff_t offset(eve::as_<T> const&) const
     {
       return 0;
```
Comment (DenisYaroshevskiy) [inline include/eve/conditional.hpp:135]: This 0 is iffy. But should work

**Review by DenisYaroshevskiy — APPROVED** [review]
I don't like that we don't have any tests. Will add some for top_bits that cover this.

File: include/eve/conditional.hpp (hunk 1 above)
Comment (jfalcou) [reply include/eve/conditional.hpp:135]: > I don't like that we don't have any tests. Will add some for top_bits that cover this.

Seems better in top_bits IMHO

**Comment by jfalcou** [comment]
> I don't like that we don't have any tests. Will add some for top_bits that cover this.

Seems better in top_bits IMHO

**Pull request merged**